app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt

```diff
@@ -170,4 +164,101 @@ open class UserProfileActivity : MyAppCompatActivity() {
                 .build()
         }
     }
+
+    override fun onPick(choice: MergeDismissImportPlaylistDialog.Choice) {
+        when (choice){
+            MergeDismissImportPlaylistDialog.Choice.MERGE -> mergePlaylist()
+            MergeDismissImportPlaylistDialog.Choice.IMPORT -> importPlaylist()
+            MergeDismissImportPlaylistDialog.Choice.DISMISS_ONLINE -> dismissOnlinePlaylist()
+        }.addOnSuccessListener {
+            user.save(this)
+        }
+
+    }
+
+    private fun mergePlaylist() : Task<Unit>{
+        Log.w(ContentValues.TAG, "MERGE")
+        return db.getUser(user.id).continueWith {
+            user.addPlaylists(it.result.getPlaylists())
+            user.isGoogleUser = true
+            user.name = it.result.name
+            user.id = it.result.id
+            db.setUser(user.id,user)
+        }
+    }
+    private fun importPlaylist() : Task<Unit>{
+        Log.w(ContentValues.TAG, "IMPORT")
+        user.removePlaylists(user.getPlaylists())
+        return db.getUser(user.id).continueWith{
+            user = it.result
+            user.name = it.result.name
+            user.id = it.result.id
+            user.isGoogleUser = true
+        }
+    }
+
+    private fun dismissOnlinePlaylist() : Task<Unit>{
+        Log.w(ContentValues.TAG, "DISMISS")
+        return db.getUser(user.id).continueWith{
+            user.name = it.result.name
+            user.id = it.result.id
+            user.isGoogleUser = true
+            db.setUser(user.id,user)
+        }
+    }
+
+    override fun onPick(choice: KeepOrDismissPlaylistDialog.Choice) {
+        when (choice){
+            KeepOrDismissPlaylistDialog.Choice.KEEP -> keepPlaylist()
+            KeepOrDismissPlaylistDialog.Choice.DISMISS -> deletePlaylist()
+        }.addOnSuccessListener{
+            User.createDefaultUser().continueWith {
+                val playlist = user.getPlaylists()
+                user = it.result
+                user.addPlaylists(playlist)
+                user.save(this)
+            }
+        }
+
+    }
+
+    private fun keepPlaylist() : Task<Unit>{
+        Log.w(ContentValues.TAG, "KEEP")
+        return Tasks.forResult(null)
+        //when keeping playlist, decide whether only those saved locally should be kept or all of them
+
+    }
+    private fun deletePlaylist() : Task<Unit> {
+        user.removePlaylists(user.getPlaylists())
+        Log.w(ContentValues.TAG, "DELETE")
```

Comment:
Do you want to keep this logging message (and previous ones) ?

Reply:
No I will delete them.